src/main/java/org/jetbrains/kotlinx/lincheck/strategy/ManagedStrategyTransformer.java

```diff
@@ -80,58 +111,667 @@ public void visitSource(String source, String debug) {
+
+        if (isSynchronized) {
+            // synchronized method is replaced with synchronized lock
+            mv = new SynchronizedBlockAddingTransformer(mname, new GeneratorAdapter(mv, access, mname, desc), className, access, classVersion);
+        }
+        mv = new UnsafeTransformer(new GeneratorAdapter(mv, access, mname, desc));
+
+        boolean shouldBeIgnored = "<clinit>".equals(mname) || shouldBeIgnored(className);
+        if (shouldBeIgnored) {
+            // static initialization sections and classes corresponding to ignoredEntryPoints
+            // should not have inserted strategy methods.
+            mv = new IgnoredSectionTransformer(mname, new GeneratorAdapter(mv, access, mname, desc));
+        } else {
+            mv = new WaitNotifyTransformer(mname, new GeneratorAdapter(mv, access, mname, desc));
+            mv = new ParkUnparkTransformer(mname, new GeneratorAdapter(mv, access, mname, desc));
+            // SharedVariableAccessMethodTransformer should be an earlier visitor than ClassInitializationTransformer
+            // not to have suspension points before 'beforeClassInitialization' call in <clinit> block.
+            mv = new LocalObjectManagingTransformer(mname, new GeneratorAdapter(mv, access, mname, desc));
+            mv = new SharedVariableAccessMethodTransformer(mname, new GeneratorAdapter(mv, access, mname, desc));
+            mv = new TimeStubTransformer(new GeneratorAdapter(mv, access, mname, desc));
+            mv = new ThreadLocalRandomTransformer(className, new GeneratorAdapter(mv, access, mname, desc));
+        }
         mv = new TryCatchBlockSorter(mv, access, mname, desc, signature, exceptions);
         return mv;
     }
 
-    class SharedVariableAccessMethodTransformer extends ManagedStrategyMethodVisitor {
-        public SharedVariableAccessMethodTransformer(String methodName, GeneratorAdapter mv) {
+    private boolean shouldBeIgnored(String className) {
+        for (String entryPoint : ignoredEntryPoints)
+            if (className.startsWith(entryPoint))
+                return true;
+        return false;
+    }
+
+    /**
+     * Changes package of transformed classes from java/util package, excluding some
+     */
+    private static class JavaUtilRemapper extends Remapper {
+        @Override
+        public String map(String name) {
+            boolean isException = name.endsWith("Exception");
+            boolean isTrustedAtomicPrimitive = TrustedAtomicPrimitives.INSTANCE.isTrustedPrimitive(name);
+            // function package is not transformed, because AFU uses it and thus there will be transformation problems
+            if (name.startsWith("java/util/") && !isTrustedAtomicPrimitive && !isException)
+                name = TRANSFORMED_PACKAGE_INTERNAL_NAME + name;
+            return name;
+        }
+    }
+
+    /**
+     * Generates body of a native method VMSupportsCS8
```

Comment:
Please add a detailed comment on why we do need this visitor, it does not seem obvious.